File: backend/services/footballAPI.js
const axios = require('axios');

// API-Football (api-football.com) — free tier: 100 calls/day
const BASE = 'https://v3.football.api-sports.io';
const KEY  = process.env.API_FOOTBALL_KEY || '';

// Cache to avoid hitting rate limits
let cache = { data: null, ts: 0 };
const CACHE_TTL = 60000; // 1 minute

async function getLiveFootball() {
  if (!KEY) return getMockFootball();
  if (Date.now() - cache.ts < CACHE_TTL && cache.data) return cache.data;

  try {
    const res = await axios.get(`${BASE}/fixtures`, {
      headers: { 'x-apisports-key': KEY },
      params: { live: 'all' },
      timeout: 8000,
    });
    const fixtures = res.data?.response || [];
    const mapped = fixtures.slice(0, 10).map(f => ({
      id:         `ff_${f.fixture.id}`,
      source:     'api_football',
      gameType:   'football_real',
      icon:       '⚽',
      label:      f.league.name,
      home:       f.teams.home.name,
      away:       f.teams.away.name,
      scoreHome:  f.goals.home ?? 0,
      scoreAway:  f.goals.away ?? 0,
      minute:     f.fixture.status.elapsed ?? 0,
      displayTime:`${f.fixture.status.elapsed ?? 0}'`,
      status:     f.fixture.status.short === 'FT' ? 'finished' : 'live',
      venue:      f.fixture.venue?.name,
      league:     f.league.name,
    }));
    cache = { data: mapped, ts: Date.now() };
    return mapped;
  } catch (e) {
    console.error('[FOOTBALL_API] Error:', e.message);
    return getMockFootball();
  }
}

async function syncFootballScores() {
  // Returns array of score updates to broadcast
  const matches = await getLiveFootball();
  return matches.map(m => ({ ...m, matchId: m.id }));
}

function getMockFootball() {
  return [
    { id:'mock_1', source:'mock', gameType:'football_real', icon:'⚽', label:'Premier League',
      home:'Man City', away:'Arsenal', scoreHome:2, scoreAway:1, minute:67,
      displayTime:"67'", status:'live' },
    { id:'mock_2', source:'mock', gameType:'football_real', icon:'⚽', label:'La Liga',
      home:'Barcelona', away:'Real Madrid', scoreHome:1, scoreAway:1, minute:55,
      displayTime:"55'", status:'live' },
  ];
}

module.exports = { getLiveFootball, syncFootballScores };

File: BETELITE_v3/BETELITE_v3/backend/services/football_api.go
package services

import (
	"encoding/json"
	"fmt"
	"net/http"
	"time"

	"betelite-go/config"
)

type FixtureData struct {
	Fixture struct {
		ID        int    `json:"id"`
		Date      string `json:"date"`
		Status    struct {
			Short string `json:"short"`
		} `json:"status"`
	} `json:"fixture"`
	Teams struct {
		Home struct {
			Name string `json:"name"`
			Logo string `json:"logo"`
		} `json:"home"`
		Away struct {
			Name string `json:"name"`
			Logo string `json:"logo"`
		} `json:"away"`
	} `json:"teams"`
	Goals struct {
		Home int `json:"home"`
		Away int `json:"away"`
	} `json:"goals"`
}

// FetchLiveFixtures fetches live football matches from API-Football
func FetchLiveFixtures() ([]FixtureData, error) {
	if config.Cfg.APIFootballKey == "" {
		return nil, fmt.Errorf("Football API Key not configured")
	}

	url := "https://v3.football.api-sports.io/fixtures?live=all"
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		return nil, err
	}
	req.Header.Set("x-apisports-key", config.Cfg.APIFootballKey)

	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var res struct {
		Response []FixtureData `json:"response"`
	}
	if err := json.NewDecoder(resp.Body).Decode(&res); err != nil {
		return nil, err
	}

	return res.Response, nil
}
